src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -414,51 +415,60 @@ void UserInfoPopup::installEvents()
-                        {
-                            if (result == UnignoreResult_Failed)
-                            {
-                                *ignoreNext = true;
-                                this->ui_.ignore->setChecked(true);
-                            }
-                            this->ui_.ignore->setEnabled(true);
-                        }
-                    });
-            }
-        });
+    QObject::connect(this->ui_.ignore, &QCheckBox::stateChanged,
+                     [this, ignoreNext, hack](int) mutable {
+                         if (*ignoreNext)
+                         {
+                             *ignoreNext = false;
+                             return;
+                         }
+
+                         this->ui_.ignore->setEnabled(false);
+
+                         auto currentUser =
+                             getApp()->accounts->twitch.getCurrent();
+                         if (this->ui_.ignore->isChecked())
+                         {
+                             getHelix()->blockUser(
+                                 this->userId_,
+                                 [this, hack, currentUser] {
+                                     this->channel_->addMessage(
+                                         makeSystemMessage("ignore success"));
+                                     if (hack.lock())
+                                     {
+                                         this->ui_.ignore->setEnabled(true);
+                                     }
+                                     currentUser->loadIgnores();
+                                 },
+                                 [this, ignoreNext] {
+                                     this->channel_->addMessage(
+                                         makeSystemMessage("ignore fail"));
+                                     *ignoreNext = true;
+                                     this->ui_.ignore->setChecked(false);
+                                 });
+                         }
+                         else
+                         {
+                             getHelix()->unblockUser(
+                                 this->userId_,
+                                 [this, hack, currentUser] {
+                                     this->channel_->addMessage(
+                                         makeSystemMessage("unignore success"));
+                                     if (hack.lock())
+                                     {
+                                         this->ui_.ignore->setEnabled(true);
+                                     }
+                                     currentUser->loadIgnores();
+                                 },
+                                 [this, ignoreNext] {
+                                     ;
+                                     this->channel_->addMessage(
+                                         makeSystemMessage("unignore fail"));
```

Comment:
maybe make the system messages be in line with the /ignore and /unignore commands